XenModel/Network/XenConnection.cs

```diff
@@ -642,7 +642,7 @@ public void WaitFor(Func<bool> predicate, Func<bool> cancelling)
             }
             try
             {
-                while (true)
+                for (int i = 0; i < 120; i++)

```

Comment:
I changed this one to behave like the WaitForCache calls and time out after 1min instead of relying on the user to cancel it. The only other places where WaitFor is used is for renaming tags and setting folders, which normally finish quite fast - according to my testing at least - so the change should be ok.